src/main/java/net/sf/jabref/gui/NewFileDialogs.java

```diff
@@ -0,0 +1,161 @@
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.filechooser.FileNameExtensionFilter;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.preferences.JabRefPreferences;
+
+/**
+ * WIP: Will replaced the FileDialogs class
+ *
+ *
+ */
+public class NewFileDialogs {
+
+    /**
+     * Custom confirmation dialog
+     * http://stackoverflow.com/a/3729157
+     */
+    private final JFileChooser fileChooser = new JFileChooser() {
+
+        @Override
+        public void approveSelection() {
+            File f = getSelectedFile();
+            if (f.exists() && (getDialogType() == SAVE_DIALOG)) {
+                int result = JOptionPane.showConfirmDialog(this, "The file exists, overwrite?", "Existing file",
+                        JOptionPane.YES_NO_CANCEL_OPTION);
+                switch (result) {
+                case JOptionPane.YES_OPTION:
+                    super.approveSelection();
+                    return;
+                case JOptionPane.NO_OPTION:
+                    return;
+                case JOptionPane.CLOSED_OPTION:
+                    return;
+                case JOptionPane.CANCEL_OPTION:
+                    cancelSelection();
+                    return;
+                }
+            }
+            super.approveSelection();
+        }
+    };
+
+    private final JFrame parent;
+    private final String directory;
+    private FileNameExtensionFilter extFilter;
+    private Set<FileExtensions> extensions = EnumSet.noneOf(FileExtensions.class);
+
+
+    public NewFileDialogs(JFrame owner) {
+        this(owner, getWorkingDir());
+    }
+
+    public NewFileDialogs(JFrame owner, String dir) {
+        this.parent = owner;
+        this.directory = dir;
+        System.out.println("DIR in Constructor" + Paths.get(dir));
+        fileChooser.setCurrentDirectory(Paths.get(dir).toFile());
+    }
+
+    private void saveFileDialog() {
+        fileChooser.setDialogType(JFileChooser.SAVE_DIALOG);
+    }
+
+    private void openFileDialog() {
+        fileChooser.setDialogType(JFileChooser.OPEN_DIALOG);
+
+    }
+
+    public NewFileDialogs dirsOnly() {
+        fileChooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+        return this;
+    }
+
+    public NewFileDialogs withExtension(FileExtensions singleExt) {
+        withExtension(EnumSet.of(singleExt));
+        return this;
+    }
+
+    public NewFileDialogs withExtension(Set<FileExtensions> fileExtensions) {
+        this.extensions = fileExtensions;
+
+        for (FileExtensions ext : fileExtensions) {
+            extFilter = new FileNameExtensionFilter(ext.getDescription(), ext.getExtensions());
+            fileChooser.addChoosableFileFilter(extFilter);
+
+        }
+
+        return this;
+    }
+
+    public NewFileDialogs updateWorkingDirectory() {
```

Comment:
Yes you are right that you can't chain in it. But in for this method, I think chained code is harder to understand.
I have no idea what `updateWorkingDirectory` does here (since it looks like that the working directory of the dialog is updated, but thats not the case):
`new FileDialog(...).withExtensions(...).updateWorkingDir().showDialog()` 
versus
````
FileDialog dialog = new FileDialog(...).withExtensions(...);
dialog.updateWorkingDir();
dialog.showDialog();
````
is more understandable in my opinion. But I have no strong opinion on this.